### PR TITLE
feat: support passing in device and client to XLA

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: lib/MLDataDevices/src/public.jl [hunk 1]
```diff
@@ -9,8 +9,10 @@ end
 struct MetalDevice <: AbstractGPUDevice end
 struct oneAPIDevice <: AbstractGPUDevice end
 
-# TODO: Later we might want to add the client field here?
-struct XLADevice <: AbstractAcceleratorDevice end
+@kwdef struct XLADevice{C, D} <: AbstractAcceleratorDevice
```
Comment (wsmoses) [inline lib/MLDataDevices/src/public.jl:12]: For future extensibility [as we get distributed and other things setup], can this be renamed to ReactantDevice?

Comment (avik-pal) [reply lib/MLDataDevices/src/public.jl:12]: yeah I can do it with a `const XLADevice = ReactantDevice` to preserve backward compatibility.

Comment (wsmoses) [reply lib/MLDataDevices/src/public.jl:12]: Sounds good!

**Comment by avik-pal** [comment]
I want to wait for https://github.com/EnzymeAD/Reactant.jl/pull/222 which exposes `device` so that we don't have to make direct XLA calls

**Pull request merged**